app/src/main/java/de/westnordost/streetcomplete/quests/construction/MarkCompletedConstruction.java

```diff
@@ -0,0 +1,80 @@
+package de.westnordost.streetcomplete.quests.construction;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import de.westnordost.osmapi.map.data.Element;
+import de.westnordost.streetcomplete.data.meta.OsmTaggings;
+import de.westnordost.streetcomplete.data.osm.Countries;
+import de.westnordost.streetcomplete.data.osm.OsmElementQuestType;
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder;
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao;
+import de.westnordost.streetcomplete.quests.AbstractQuestAnswerFragment;
+import de.westnordost.streetcomplete.quests.DateUtil;
+import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment;
+
+public abstract class MarkCompletedConstruction implements OsmElementQuestType {
+	protected final OverpassMapDataDao overpassServer;
+
+	MarkCompletedConstruction(OverpassMapDataDao overpassServer)
+	{
+		this.overpassServer = overpassServer;
+	}
+
+	@Nullable
+	@Override public Boolean isApplicableTo(Element element)
+	{
+		/* Whether this element applies to this quest cannot be determined by looking at that
+		   element alone (see download()), an Overpass query would need to be made to find this out.
+		   This is too heavy-weight for this method so it always returns false. */
+
+		/* The implications of this are that this quest will never be created directly
+		   as consequence of solving another quest and also after reverting an input,
+		   the quest will not immediately pop up again. Instead, they are downloaded well after an
+		   element became fit for this quest. */
+		return null;
+	}
+
+	public AbstractQuestAnswerFragment createForm()
+	{
+		return new YesNoQuestAnswerFragment();
+	}
+
+	protected String getCurrentDateString(){
+		return DateUtil.getCurrentDateString() + "T00:00:00Z";
+	}
+
+	protected String getOffsetDateString(int offset){
+		return DateUtil.getOffsetDateString(offset) + "T00:00:00Z";
+	}
+
+	String getQueryPart(String key, String nameOfGeneratedGroup, int reviewIntervalInDays){
+		return "[" + key + "=construction]" +
+			"(if:!is_date(t['opening_date']) || date(t['opening_date'])<date('" + getCurrentDateString() + "'))" +
+			" -> .construction_with_unknown_state; " +
+			"way[" + key + "=construction](newer: '" + getOffsetDateString(-reviewIntervalInDays) +"') -> .recently_edited_construction; " +
+			"(.construction_with_unknown_state - .recently_edited_construction) -> " + nameOfGeneratedGroup + ";";
+	}
+
+	void removeTagsDescribingConstruction(StringMapChangesBuilder changes) {
+		changes.deleteIfExists("construction");
+		changes.deleteIfExists("source:construction");
+		changes.deleteIfExists("opening_date");
+		changes.deleteIfExists("source:opening_date");
+		changes.deleteIfExists(OsmTaggings.SURVEY_MARK_KEY);
+		changes.deleteIfExists("source:" + OsmTaggings.SURVEY_MARK_KEY);
```

Comment:
Should the app not also understand the other [`check_date`-like keys](https://wiki.openstreetmap.org/wiki/Key:check_date) as long as they are still around in mass?

Especially `survey:date` should be understood as it seems to be even more popular than `check_date`, even minus the Antarctica import. What are the reasons why you decided for `check_date` instead of `survey:date`? (Perhaps these reasons could be documented in the wiki to get more people to agree on one tagging)

Reply:
(This also means that they should turn up in the query)

Reply:
> (This also means that they should turn up in the query)

Not necessary, all survey tags are supported as only ways without edits for specified time are selected.

Reply:
> What are the reasons why you decided for check_date instead of survey:date? 

From what I remember - because it was already documented on wiki.

Reply:
Replying to comment "I don't understand" that github is showing elsewhere (it got confused by reply using mail)

Currently road construction will not show any objects edited within last 14 days.

As result any edit updating/adding check_date, survey:date or any other tag will cause element to be not eligible for this quest for 14 days.

Main intended effect is that any survey marker tag is supported without need to parse tags.

Reply:
Ah I see. Okay.

Reply:
I added explanation in a comment as apparently it is not obvious.